checker/src/main/java/org/checkerframework/checker/resourceleak/MustCallConsistencyAnalyzer.java

```diff
@@ -0,0 +1,1705 @@
+        // TODO: Getting this every time is inefficient if a method has many @CreatesMustCallFor
+        // annotations, but that should be a rare path.
+        MethodTree enclosingMethodTree = TreePathUtil.enclosingMethod(currentPath);
+        if (enclosingMethodTree != null) {
+          ExecutableElement enclosingMethodElt =
+              TreeUtils.elementFromDeclaration(enclosingMethodTree);
+          MustCallAnnotatedTypeFactory mcAtf =
+              typeFactory.getTypeFactoryOfSubchecker(MustCallChecker.class);
+          List<String> enclosingCmcfValues =
+              ResourceLeakVisitor.getCreatesMustCallForValues(
+                  enclosingMethodElt, mcAtf, typeFactory);
+          if (!enclosingCmcfValues.isEmpty()) {
+            for (String enclosingCmcfValue : enclosingCmcfValues) {
+              JavaExpression enclosingTarget;
+              try {
+                enclosingTarget =
+                    StringToJavaExpression.atMethodBody(
+                        enclosingCmcfValue, enclosingMethodTree, checker);
+              } catch (JavaExpressionParseException e) {
+                // Do not issue an error here, because it would be a duplicate.
+                // The error will be issued by the Transfer class of the checker,
+                // via the CreatesMustCallForElementSupplier interface.
+                enclosingTarget = null;
+              }
+
+              if (representSame(target, enclosingTarget)) {
+                // this satisifies case 3
+                validInvocation = true;
+              }
+            }
+          }
+        }
+      }
+      if (!validInvocation) {
+        missing.add(target);
+      }
+    }
+
+    if (missing.isEmpty()) {
+      // all targets were valid
+      return;
+    }
+
+    String missingStrs = StringsPlume.join(", ", missing);
+    checker.reportError(node.getTree(), "reset.not.owning", missingStrs);
+  }
+
+  /**
+   * Checks whether the two JavaExpressions are the same. This is identical to calling equals() on
+   * one of them, with two exceptions: the second expression can be null, and "this" references are
+   * compared using their underlying type. (ThisReference#equals always returns true, which isn't
+   * accurate in the case of nested classes.)
+   *
+   * @param target a JavaExpression
+   * @param enclosingTarget another, possibly null, JavaExpression
+   * @return true iff they represent the same program element
+   */
+  private boolean representSame(JavaExpression target, @Nullable JavaExpression enclosingTarget) {
+    if (enclosingTarget == null) {
+      return false;
+    }
+    if (enclosingTarget instanceof ThisReference && target instanceof ThisReference) {
+      return enclosingTarget.getType().toString().equals(target.getType().toString());
+    } else {
+      return enclosingTarget.equals(target);
+    }
+  }
+
+  /**
+   * Given a node representing a method or constructor call, checks that if the call has a non-empty
```

Comment:
"if the call has": is the check of the return type of the called method, or of the type of the call expression?